chain/client/src/client.rs

```diff
@@ -620,11 +621,19 @@ impl Client {
             }
         }
 
-        for missing_chunks in blocks_missing_chunks.write().unwrap().drain(..) {
-            self.shards_mgr.request_chunks(missing_chunks).unwrap();
-        }
+        // Request any missing chunks
+        let requests_result = self.shards_mgr.request_chunks(
```

Comment:
We should not fail to accept the current block because we weren't able to fetch chunks for *some other* block.
Possibly this error should be propagated as an extra return value?

Reply:
I thought about including an extra return value, but it seemed like the existing `Result` in the tuple was already used to signal failure in various ways (there are several tests which match on it). So I figured chaining this with the existing one would be enough to propagate the error. I could try to add a specific test for this if you like. Whether or not I can contrive one is perhaps a good exercise in general given @bowenwang1996 's comment about whether or not this can happen at all.